src/components/settings/LocationSettings.js

```diff
@@ -0,0 +1,163 @@
+import * as Location from 'expo-location';
+import React, { useEffect, useState } from 'react';
+import { Alert, View } from 'react-native';
+import Collapsible from 'react-native-collapsible';
+
+import { colors, texts } from '../../config';
+import { useLocationSettings } from '../../hooks';
+import { WebViewMap } from '../map';
+import { Button } from '../Button';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { RegularText } from '../Text';
+import { ToggleListItem } from '../ToggleListItem';
+import { Wrapper, WrapperWithOrientation } from '../Wrapper';
+import { location, locationIconAnchor } from '../../icons';
+import { latLngToLocationObject } from '../../helpers';
+
+const baseLocationMarker = {
+  icon: location(colors.primary),
+  iconAnchor: locationIconAnchor
+};
+
+const useSystemPermission = () => {
+  const [systemPermission, setSystemPermission] = useState();
+
+  useEffect(() => {
+    (async () => setSystemPermission(await Location.getForegroundPermissionsAsync()))();
+  }, []);
+
+  return systemPermission;
+};
+
+const getLocationMarker = (locationObject) => ({
+  ...baseLocationMarker,
+  id: 'alternativePosition',
```

Comment:
the `id` is used for handling map marker events. i thought that the id was mandatory for that reason, but when double checking it turned out to be optional. since we do not use it, i removed it from both places